File: medium/product-of-array-except-self/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res(size(nums),1);
        int sum=1;
        for (int i=0; i<size(nums); ++i) {
            res[i]*=sum;
            sum*=nums[i];
        }
        sum=1;
        for (int i=size(nums)-1; i>-1; --i) {
            res[i]*=sum;
            sum*=nums[i];
        }
        return res;
    }
};

File: medium/product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res=[1]*len(nums)
        sumsum=1
        for i in range(0,len(nums),1):
            res[i]*=sumsum
            sumsum*=nums[i]
        sumsum=1
        for i in range(len(nums)-1,-1,-1):
            res[i]*=sumsum
            sumsum*=nums[i]
        return res
